cns/requestcontroller/kubernetes/k8srequestcontroller.go

```diff
@@ -0,0 +1,175 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const k8sNamespace = "kube-system"
+
+// k8sRequestController watches CRDs for status updates and publishes CRD spec changes
+// mgr acts as the communication between requestController and API server
+// implements the RequestController interface
+type k8sRequestController struct {
+	mgr         manager.Manager             //mgr has method GetClient() to get k8s client
+	Client      K8sClient                   //Relying on the K8sClient interface to more easily test
+	restService *restserver.HTTPRestService //restService is given to nodeNetworkConfigReconciler (the reconcile loop)
```

Comment:
why does the reconcile loop need a server?

Reply:
This server is actually CNS's state, so reconcile will change the cns's state directly based on the incoming status